javascript/extractor/src/com/semmle/js/extractor/DependencyInstallationResult.java

```diff
@@ -6,24 +6,39 @@
 
 /** Contains the results of installing dependencies. */
 public class DependencyInstallationResult {
+  private Path sourceRoot;
   private Path virtualSourceRoot;
   private Map<String, Path> packageEntryPoints;
   private Map<String, Path> packageJsonFiles;
 
   public static final DependencyInstallationResult empty =
-      new DependencyInstallationResult(null, Collections.emptyMap(), Collections.emptyMap());
+      new DependencyInstallationResult(null, null, Collections.emptyMap(), Collections.emptyMap());
 
   public DependencyInstallationResult(
+      Path sourceRoot,
       Path virtualSourceRoot,
       Map<String, Path> packageEntryPoints,
       Map<String, Path> packageJsonFiles) {
+    this.sourceRoot = sourceRoot;
+    this.virtualSourceRoot = virtualSourceRoot;
     this.packageEntryPoints = packageEntryPoints;
     this.packageJsonFiles = packageJsonFiles;
   }
 
+  /**
+   * Returns the source root mirrored by {@link #getVirtualSourceRoot()} or <code>null</code>
+   * if no virtual source root exists.
+   * <p>
+   * When invoked from the AutoBuilder, this corresponds to the source root. When invoked
+   * from ODASA, there is no notion of source root, so this is always <code>null</code> in that context.
+   */
+  public Path getSourceRoot() {
+    return sourceRoot;
+  }
+
   /**
    * Returns the virtual source root or <code>null</code> if no virtual source root exists.
-   *
+   * <p>
```

Comment:
```suggestion
   *
```

There is a whole bunch of the `<p>` throughout the documentation, and none of them have an corresponding `</p>`. 
You don't have to fix this one, because there is a bunch more throughout our `.java` files.

Reply:
The `</p>` is not needed and our auto-formatter removes them.

Without the `<p>` javadoc won't create a separate paragraph for the following text, not even a newline.